src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,348 @@
+                for (int i = 0; i < value.Length; i += chunkSize)
+                {
+                    int remainingBytes = Math.Min(chunkSize, value.Length - i);
+                    bool isFinalBlock = false ;
+
+                    // Take a chunk of bytes from the input value.
+                    ReadOnlyMemory<byte> chunk = value.Slice(i, remainingBytes);
+
+                    // If the buffer is not empty, then we copy the bytes from the buffer
+                    // to the current chunk being processed.
+                    if (numBytesNotWrittenFromPreviousChunk > 0)
+                    {
+                        ReadOnlyMemory<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(0,numBytesNotWrittenFromPreviousChunk);
+
+                        int totalLength = bytesNotProcessedFromPreviousChunk.Length + chunk.Length;
+
+                        // Rent an array to hold bytes from both previous and current chunks.
+                        byte[] combinedArray = ArrayPool<byte>.Shared.Rent(totalLength);
+
+                        // Copy bytes from bytesNotProcessedFromPreviousChunk to the combined array
+                        bytesNotProcessedFromPreviousChunk.Span.CopyTo(combinedArray);
+
+                        // Copy bytes from chunk to the combined array, starting from the end of bytesNotProcessedFromPreviousChunk
+                        chunk.Span.CopyTo(combinedArray.AsSpan(bytesNotProcessedFromPreviousChunk.Length));
+
+                        WriteChunk(combinedArray, totalLength, isFinalBlock);
+
+                        ArrayPool<byte>.Shared.Return(combinedArray);
+                    }
+                    else
+                    {
+                        WriteChunk(chunk.Span, chunk.Length, isFinalBlock);
+                    }
+
+                    // Flush the writer if the buffer threshold is reached
+                    await this.jsonWriter.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+                }
+            }
+
+            /// <summary>
+            /// Writes a chunk of data, encoding it using Base64, and updates the buffer with any unprocessed bytes.
+            /// </summary>
+            /// <param name="chunk">The chunk of data to be encoded and written.</param>
+            /// <param name="isFinalBlock">A boolean indicating whether this is the final block of data.</param>
+            private void WriteChunk(ReadOnlySpan<byte> chunk, int chunkLength, bool isFinalBlock)
+            {
+                // Encode the current chunk using Base64 and write it
+                this.jsonWriter.Base64EncodeAndWriteChunk(chunk.Slice(0, chunkLength), isFinalBlock, out numBytesNotWrittenFromPreviousChunk);
+
+                if (numBytesNotWrittenFromPreviousChunk > 0)
+                {
+                    if (this.buffer == null)
+                    {
+                        this.buffer = ArrayPool<byte>.Shared.Rent(chunkSize);

```

Comment:
Where is the `chunkSize` defined? Or is this the parent JsonWriter's `chunkSize`?

Reply:
I think it's hard to tell the right chunksize to use here.. Having a small chunkSize leads to `destination small error`. It is possible that the whole chunksize is not processed in a method call. That's why I resorted to using the whole chunksize that's defined in the parent writer.

Reply:
I'm surprise that you would get a whole chunkSize worth of data not being processed in a single write call. Could you share a scenario where that occurred?

That said, my initial question was based on the fact that I couldn't see where `chunkSize` is defined. If it's the parent JsonWriter's chunkSize, then maybe it would be clearer to use `this.jsonWriter.chunkSize`.